### PR TITLE
Update URL handling to use yarl

### PR DESCRIPTION
**Pull request by bdraco**
aiohttp uses yarl for url handling. Remove the urllib warning disables since they are unused and switch all urllib using to yarl

### CODE REVIEW

**Comment by jrester** [comment]
@bdraco thanks for your contribution, but currently one test is failing locally. Please fix this test, than this PR can be merged.

File: tesla_powerwall/api.py [hunk 1]
```diff
@@ -21,10 +19,7 @@ def __init__(
         verify_ssl: bool = False,
         disable_insecure_warning: bool = True,
     ) -> None:
-        if disable_insecure_warning:
-            disable_warnings(InsecureRequestWarning)
-
-        self._endpoint = self._parse_endpoint(endpoint)
+        self._endpoint = URL(self._parse_endpoint(endpoint))
```
Comment (jrester) [inline tesla_powerwall/api.py:22]:
```suggestion
        self._endpoint = URL(endpoint).with_scheme("https").with_path("api")
```

yarl makes the URL handling much simpler and makes the `_parse_endpoint` method obsolete.

File: tesla_powerwall/api.py [hunk 2]
```diff
@@ -21,10 +19,7 @@ def __init__(
         verify_ssl: bool = False,
         disable_insecure_warning: bool = True,
```
Comment (jrester) [inline tesla_powerwall/api.py:20]:
```suggestion
```

This is no longer needed then. Please also update the documentation in the `README.md` accordingly.

**Review by jrester — CHANGES_REQUESTED** [review]

File: tesla_powerwall/api.py [hunk 3]
```diff
@@ -19,12 +17,10 @@ def __init__(
...
         http_session: Optional[aiohttp.ClientSession] = None,
         verify_ssl: bool = False,
-        disable_insecure_warning: bool = True,
     ) -> None:
-        if disable_insecure_warning:
-            disable_warnings(InsecureRequestWarning)
-
-        self._endpoint = self._parse_endpoint(endpoint)
+        if not endpoint.startswith("http"):
+            endpoint = f"https://{endpoint}"
```
Comment (bdraco) [inline tesla_powerwall/api.py:22]: YARL will refuse to change schema for it doesn't already have one

Comment (bdraco) [reply tesla_powerwall/api.py:22]: The HA integration passes only the address so I didn't want to make this breaking

**Pull request merged**